### PR TITLE
Add Compressed Linux zImage support

### PR DESCRIPTION
**Pull request by ninjahacker**
This is to make the initial support for compressed Linux kernels as discussed in issue #2316. Not many features have been added yet, but it is a start for identification of embedded ARM Linux kernels.

### CODE REVIEW

File: libr/bin/format/zimg/zimg.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/* radare - LGPL - Copyright 2009-2015 - pancake */
+
+#include <r_types.h>
+#include <r_util.h>
+#include "zimg.h"
+
+
+#define FAIL(x) { eprintf(x); goto fail; }
```
Comment (crowell) [inline libr/bin/format/zimg/zimg.c]: this macro is never used?

**Comment by ninjahacker** [comment]
That's correct. Sorry. I thought it would come up, but then I forgot about it.

File: libr/bin/format/zimg/zimg.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#include <r_types.h>
+#include <r_util.h>
+#include <r_lib.h>
+#include <r_bin.h>
+
+#define R_BIN_ZIMG_MAXSTR 256
+
+#pragma pack(4)
```
Comment (crowell) [inline libr/bin/format/zimg/zimg.h]: what's the purpose of packing this. should you push then pop this pack pragma? like is done here?
http://www.cplusplus.com/forum/general/14659/

File: libr/bin/format/zimg/zimg.c [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/* radare - LGPL - Copyright 2009-2015 - pancake */
```
Comment (crowell) [inline libr/bin/format/zimg/zimg.c]: credit yourself on the new files! :+1:

File: libr/bin/p/bin_zimg.c [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	.init = NULL,
+	.fini = NULL,
+	.get_sdb = &get_sdb,
+	.load = &load,
+	.load_bytes = &load_bytes,
+	.destroy = NULL,
+	.check = &check,
+	.check_bytes = &check_bytes,
+	.baddr = &baddr,
+	.boffset = NULL,
```
Comment (radare) [inline libr/bin/p/bin_zimg.c]: No need to define all those NULLs.

**Comment by radare** [comment]
lgtm

**Comment by radare** [comment]
merged :)

**Comment by Maijin** [comment]
@radare Why not wait CI to tests before merging :< ?

**Comment by radare** [comment]
Life is too short

**Comment by jjdredd** [comment]
so make it longer by fixing new bugs

**Pull request closed without merging**